### PR TITLE
Allow empty block

### PR DESCRIPTION
**Pull request by zatan**

### CODE REVIEW

**Comment by tomkins** [comment]
Good stuff!

One problem I'm getting is that if you edit a page, add an empty block, save a version, then try and edit it - the creation of a new version currently fails.

If you could add or edit an existing test for it.

I think that's the only issue.

**Comment by zatan** [comment]
Updated

File: glitter/page.py [hunk 1]
```diff
@@ -39,15 +39,23 @@ def render(self, rerender):
         # Add some classes to the block to help style it
         block_classes = self.css_classes()
 
-        # Render the block
```
Comment (tomkins) [inline glitter/page.py]: Think this needs changing. If the block doesn't exist, we still probably want to use the blocks chosen render function.

Something like:

```
block_class = self.content_block.content_type.model_class()
render_function = block_class.render_function
```

Only downside... we'll have to check all existing blocks if they've got a custom render function.

File: glitter/static/glitter/js/glitter_editor.js [hunk 2]
```diff
@@ -269,6 +274,23 @@ GlitterEditor.jQuery = jQuery.noConflict(true);
             });
         });
 
+        $(document).on('click', '.glitter-delete-block-ajax', function(e){
+          e.preventDefault();
+          var ajax_url = $(this).attr('data-ajax-url');
+
+          // Delete element.
+          $(this).closest('.block-header').remove();
```
Comment (tomkins) [inline glitter/static/glitter/js/glitter_editor.js]: An empty block might still produce some content (doesn't need configuring, etc.)

Try:

```
var block_header = $(this).closest('.block-header');
block_header.next('.glitter_page_block').remove();
block_header.remove();
```

**Comment by tomkins** [comment]
:shipit:

**Pull request merged**